src/sentry/models/release.py

```diff
@@ -937,3 +962,32 @@ def get_artifact_counts(release_ids: List[int]) -> Mapping[int, int]:
     )
     qs.query.group_by = ["release_id"]
     return dict(qs)
+
+
+def follows_semver_versioning_scheme(org_id, project_id, release_version=None):
+    """
+    Checks if we should follow semantic versioning scheme for ordering based on
+    1. Latest ten releases of the project_id passed in all follow semver
+    2. provided release version argument is a valid semver version
+
+    Inputs:
+        * org_id
+        * project_id
+        * release_version
+    Returns:
+        Boolean that indicates if we should follow semantic version or not
+    """
+    # ToDo(ahmed): Replace check for lastest 10 releases by check for project_options once it is
+    #  implemented & move function else where to be easily accessible for re-use
+    follows_semver = True
+
+    # Check if the latest ten releases are semver compliant
+    releases_list = Release.objects.filter(
+        organization=org_id, projects__id__in=[project_id]
+    ).order_by("-date_added")[:10]
+    follows_semver = follows_semver and all(release.is_semver_release for release in releases_list)
```

Comment:
I am skeptical about `any` because what if only one coincidentally matches semver maybe we can change it to atleast like 7/10 of them have to be semver?

Reply:
It's kind of tricky because there are a few cases we will handle here:
- User just started using semver, ideally we want to start marking them as using semver fairly quickly
- User made a mistake and accidentally stopped using semver a couple times
- User made a mistake and accidentally started using semver a couple times
- User stopped using semver

What if we checked that there is a semver release in the most recent 2-3 releases, as well as having at least 3(?) releases using semver in the last 10?

Reply:
I think that's a good idea 
What about if the user has less than 10 releases? How would we check for that?

Reply:
Maybe in that case we forego the next condition i.e. forego the atleast 3 releases in the last 10 and finding one semver in the last 3 should suffice.. wdyt

Reply:
Yeah, that sounds good to me!